pkg/expr/convert_to_long.go

```diff
@@ -0,0 +1,315 @@
+package expr
+
+import (
+	"fmt"
+	"sort"
+	"time"
+
+	"github.com/grafana/grafana-plugin-sdk-go/data"
+)
+
+func ConvertToLong(frames data.Frames) (data.Frames, error) {
+	if len(frames) == 0 {
+		// general empty case for now
+		return frames, nil
+	}
+	// Four Conversion Possible Cases
+	// 1. NumericMulti -> NumericLong
+	// 2. NumericWide -> NumericLong
+	// 3. TimeSeriesMulti -> TimeSeriesLong
+	// 4. TimeSeriesWide -> TimeSeriesLong
+
+	// Detect if input type is declared
+	// First Check Frame Meta Type
+
+	var inputType data.FrameType
+	if frames[0].Meta != nil && frames[0].Meta.Type != "" {
+		inputType = frames[0].Meta.Type
+	}
+	if inputType != "" && !supportedToLongConversion(inputType) {
+		return frames, fmt.Errorf("unsupported input type %s for SQL expression", inputType)
+	}
+
+	// TODO: Add some guessing of Type if not declared
+
+	if inputType == "" {
+		return frames, fmt.Errorf("could not determine input type")
+	}
+
+	convert := getToLongConversionFunc(inputType)
```

Comment:
I have a slight preference for the name `toLong` here. I think it's a little more specific.
```suggestion
	toLong := getToLongConversionFunc(inputType)
```